### PR TITLE
Configuration parsing test

### PR DESCRIPTION
**Pull request by britter**
While working on #3 I realized that I will have to change the way ReporterConfigurations are implemented at the moment. For this reason I need a test suit that makes sure configuration parsing still works as expected.
This needs to be merged before I can continue working on #3.

### CODE REVIEW

File: src/test/scala/de/bripkens/ha/ConfigurationSpec.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        val expected = HealthCheckEndpoint("http://127.0.0.1:8181/healthcheck", "recommendation", "Recommendation System", 10000, "myConsoleReporter")
+        configuration.endpoints should contain(expected)
+      }
+
+      "have two reporters" in {
+        configuration.reporters should have size 2
+      }
+
+      "have a slack reporter" in {
+        val expected = SlackReporterConfig(null, "test", "https://hooks.slack.com/services/12345", "Health Check", "http://lorempixel.com/64/64/")
```
Comment (britter) [inline src/test/scala/de/bripkens/ha/ConfigurationSpec.scala:38]: I don't understand why the type is null.

Comment (bripkens) [inline src/test/scala/de/bripkens/ha/ConfigurationSpec.scala:38]: Interesting. Seems like the Jackson sub type identification works based on the JSON property name. On the other hand, the Scala module does not seem to respect the `JsonProperty` annotation. Would be interesting to see what happens when the property name is changed from `reporterType` to just `type` in the configuration classes.

**Comment by bripkens** [comment]
I'd be okay with merging this - even with the null check as this the current behavior of the application. If you want to, feel free to inspect and change the property name to `type`. Would be helpful to understand the reason for this behavior :).

Anyhow, thank you **very** much for your help! :)

**Comment by britter** [comment]
> feel free to inspect and change the property name to `type`

Changing the implementation to:

``` scala
@JsonCreator
@JsonTypeInfo(
  use = JsonTypeInfo.Id.NAME,
  include = JsonTypeInfo.As.PROPERTY,
  property = "type"
)
@JsonSubTypes(value = Array(
  new JsonSubTypes.Type(value = classOf[SlackReporterConfig], name = "slack"),
  new JsonSubTypes.Type(value = classOf[ConsoleReporterConfig], name = "console")
))
abstract class AbstractReporterConfig(@JsonProperty("type") `type`: String) {
  val implementation: Class[_]
}

case class SlackReporterConfig(@JsonProperty("type") `type`: String,
                               @JsonProperty("channel") channel: String,
                               @JsonProperty("webhookUrl") webhookUrl: String,
                               @JsonProperty("botName") botName: String,
                               @JsonProperty("botImage") botImage: String)
  extends AbstractReporterConfig(`type`) {
  override val implementation = classOf[SlackReporter]
}

case class ConsoleReporterConfig(@JsonProperty("type") `type`: String)
  extends AbstractReporterConfig(`type`) {
  override val implementation = classOf[ConsoleReporter]
}
```

yields the same result: `type` ist still null. I think I'm okay with this for now, since I'd like to focus on #3 instead to figuring out what is going on with Jackson :-)

**Comment by bripkens** [comment]
Thanks :)

**Pull request merged**